src/sth_database.js

```diff
@@ -812,6 +852,84 @@
     });
   }
 
+  /**
+   * Generates a hash based on an input and a maximum number of bytes
+   * @param input The input to generate the hash from
+   * @param limit The maximum number of bytes of the hash
+   */
+  function generateHash(input, limit) {
+    var shasum = crypto.createHash('sha512');
+    shasum.update(input);
+    var hash = shasum.digest('hex');
+    if (limit) {
+      hash = hash.substr(0, limit);
+    }
+    return hash;
+  }
+
+  /**
+   * Stores the collection name (hash) in the database
+   * @param params The params used to generate the collection name (hash)
+   * @param hash The collection name (hash)
```

Comment:
If it is the colllection name, not the hash, I would call it `collectionName`. Just a suggestion, not important :)